docs/unit_testing.rst

```diff
@@ -0,0 +1,79 @@
+Unit testing
+############
+
+Create unit tests so that each test is a subclass of :class:`unittest.TestCase` from Python's
+`unittest <https://docs.python.org/3/library/unittest.html>`_ library as usual. A good practice is to place the tests
+into `tests` package in your root application package. And then it's all up to the developers and we provide some
+hints.
+
+There are various choices how to test and what to test regarding Zsl apps.
+
+First of all, if you do not need a ZSL instance, do not create it and run the unit tests without the instance. It will
+speed the things up. If you really need it see the section about
+:ref:`unit-testing-zsl-instance`.
+
+Running the unit tests
+======================
+
+To run all the unit tests in the package `<app_root_package>.tests` run the following.
+
+.. code-block:: console
+
+    $ cd <app_root_package>/.. # To the directory where the application root package is.
+    $ python -m unittest discover <app_root_package>.tests '*_test.py'
+
+See the `example project tests <https://github.com/AtteqCom/zsl_examples/tree/master/time_tracker/time_tracker/tests>`_
+on how to use the tests.
+
+There are various test mixins that could be used while testing.
+
+.. _unit-testing-zsl-instance:
+
+Testing with a Zsl instance
+---------------------------
+
+We show an example test case which can be reused.
+
+.. code-block:: python
+
+    class ToBeTestedTaskTestCase(ZslTestCase, HttpTestCase, TestCase):
+        ZSL_TEST_CONFIGURATION = ZslTestConfiguration(app_name='time_tracker', container=WebContainer, version=None,
+                                                      profile=None)
+
+
+Add :class:`zsl.unittest.zsl.ZslTestCase` mixin to your tests so that the `setUpClass` method is run during the test
+class initialization phase. This means `ZslTestCase` must be before `unittest.TestCase`.
```

Comment:
This is not a trivial subclassing, you can have more examples like these. We could make a much more complicated things for those. You can, if you want, I do not since the benefit is low and the time spent's not worth it.

Think about a TestCase needing two setUpClass methods. With this they can be properly called I guess.